src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -266,6 +268,24 @@ public void addExternalTrigger(ExternalTrigger external_trigger) {
 		ext_triggers.put(external_trigger.getID(), external_trigger.getName());
 	}
 
+	@SuppressWarnings("unused") public String[] getAllDependencies(String procedureName) {
+		ModElement me = mcreator.getWorkspace().getModElementByName(procedureName);
+		return me != null && me.getGeneratableElement() instanceof Procedure procedure ?
+				procedure.getDependencies().stream().map(Object::toString).toArray(String[]::new) :
+				new String[0];
+	}
+
+	@SuppressWarnings("unused") public String getDependencyType(String procedureName, String dependencyName) {
+		ModElement me = mcreator.getWorkspace().getModElementByName(procedureName);
+		if (me != null && me.getGeneratableElement() instanceof Procedure procedure) {
+			Optional<String> dependency = procedure.getDependencies().stream()
+					.filter(e -> e.getName().equals(dependencyName)).map(Dependency::getRawType).findFirst();
+			if (dependency.isPresent())
+				return VariableTypeLoader.INSTANCE.fromName(dependency.get()).getBlocklyVariableType();

```

Comment:
This method can throw a NPE, because `fromName` can return null. 
Another issue with the current approach is that it's possible to pass *any* value to a dependency that doesn't match any variable type. In this example, I'm passing a string to the `world` dependency:

![compile error](https://github.com/MCreator/MCreator/assets/64546452/0772b26f-374b-47d0-b7bf-5d38a01ae9b9)

Reply:
So I suppose I should e.g. disable the block if such unsupported dependencies are added to the custom signature?

Reply:
Possibly, or also check if the attached block has `null` output type (e.g. custom code block), so users can still pass those if they need to

Reply:
I don't think the fix was tested properly, as now the block with x y z is disabled in the side panel:
![disabled block](https://github.com/MCreator/MCreator/assets/64546452/f9019e48-5cfc-404c-ae0b-4379ef281573)

The block is also disabled when passing a value to a unused parameter, instead of ignoring it.

Reply:
Turns out blocks face the same events when being shown in toolbox and when being added to the editor.

Reply:
✔️ Fixed (I think)

Reply:
Please test those things yourself too before asking for another review. I think such a case should be caught by the PR author while validating their changes, not by the maintainer.

Reply:
AlsoSomeoneElse always finds some scenario that I miss when testing my changes. I have totally no idea why/where do I lack the skill 🤷‍♂️

Reply:
Checking toolbox works and block can still be placed from it is almost minimal testing needed before marking RFR